src/parlant/core/engines/alpha/engine.py

```diff
@@ -356,7 +356,7 @@ async def _do_process(
             ):
                 message_generation_inspections.append(
                     MessageGenerationInspection(
-                        generation=event_generation_result.generation_info,
+                        generation=event_generation_result.generations[0],
```

Comment:
It's incorrect to do `generations[0]` here. What if there were no generations? What if there was more than one? This is coincidental programming at this point.

What really happened here was that `EventGenerationResult` took on a different contract with respect to tool calls and messages. So, instead of forcing them both still under one heading, maybe we don't need to have a reusable dataclass here anymore.

Remove `event_generation.py`, add `MessageEventGenerationResult` under the message generation module, and `ToolEventGenerationResult` under the tool call generation module.